src/xin_memory_history_manager.cpp

```diff
@@ -0,0 +1,70 @@
+/***************************************************************************
+* Copyright (c) 2018, Martin Renou, Johan Mabille, Sylvain Corlay and      *
+* Wolf Vollprecht                                                          *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include <stdexcept>
+#include <string>
+
+#include "xeus/xin_memory_history_manager.hpp"
+#include "xeus/xjson.hpp"
+
+namespace xeus
+{
+    xin_memory_history_manager::xin_memory_history_manager()
+    {
+    }
+
+    xin_memory_history_manager::~xin_memory_history_manager()
+    {
+    }
+
+    void xin_memory_history_manager::configure_impl()
+    {
+    }
+
+    void xin_memory_history_manager::store_inputs_impl(int line_num, const std::string& input)
+    {
+        m_history.push_back({"0", std::to_string(line_num), input});
+    }
+
+    xjson xin_memory_history_manager::get_tail_impl(int n, bool /*raw*/, bool /*output*/) const
+    {
+        xjson reply;
+        history_type history;
+
+        int count = std::min(n, static_cast<int>(m_history.size()));
+        std::copy_n(m_history.cbegin(), count, std::back_inserter(history));
+
+        reply["status"] = "ok";
+        reply["history"] = history;
+
+        return reply;
+    }
+
+    xjson xin_memory_history_manager::get_range_impl(int /*session*/, int start, int stop, bool /*raw*/, bool /*output*/) const
+    {
+        xjson reply;
+        history_type history;
+
+        int count = std::min(stop, static_cast<int>(m_history.size())) - start;
```

Comment:
What if `start` is bigger than `stop` and `m_history.size()` ?

Reply:
I can raise a `runtime_error` in that case